src/lookup-table.c

```diff
@@ -97,34 +99,36 @@ static bool binsearch(const struct lookup_table *base, const uint32_t hash,
 			// occurrence of the key in the array
 			return true;
 		}
-		else if(nel == 1)
+		else if(size == 1)
 		{
 			// If there's only one element left and it doesn't
 			// match, break the loop as the key is not in the array
 			// => return false
 
-			// If the key is less than the middle element, the key
-			// would be inserted before the middle element
-			if(sign < 0)
-				break;
+			// If the key is greater than the middle element, the
+			// key would be inserted after the middle element
+			if (sign > 0)
+				(*try)++;
 
-			// If the key is greater than the middle element, the key
-			// would be inserted after the middle element
-			(*try)++;
+			// Break the loop, we have not found the key but
+			// know that it would need to be inserted at *try
 			break;
 		}
-		else if(sign < 0)
+		else if (sign < 0)
```

Comment:
I'm not sure if you wants to use a specific code style to format the code, but in this file I noticed you added a space after `if` statements, but not after others.

You added a space in every new line, except in line 102 or in the `while` (line 85).

All previous existent code doesn't use a space before parentheses.

Reply:
Interesting, seems I have forgotten to disable some automatic code linting. Should be force-pushed away now.